### ISSUE TITLE
Bug: Fix distributed dataset batch size allocation causing NaN RF training data

### ISSUE THREAD

**Issue by zachtheyek** [zachtheyek/Aetherscan#22]
## Summary

Fixed a critical bug in distributed dataset preparation that was silently allocating fewer samples per replica than intended, causing NaN values during RF classifier training.

## Root Cause

In `prepare_distributed_train_dataset()` and `prepare_distributed_inf_dataset()`, the dataset batching was using `per_replica_batch_size` instead of the global batch size (`per_replica_batch_size * num_replicas`).

When using TensorFlow's `experimental_distribute_dataset()`, the strategy automatically shards the dataset across replicas. By passing a smaller batch size to `.batch()`, each replica was receiving `1/num_replicas` fewer samples than expected.

**Before (buggy):**
```python
train_dataset = (
    tf.data.Dataset.from_generator(train_generator, ...)
    .batch(per_replica_batch_size, drop_remainder=True)  # Wrong!
    ...
)
```

**After (fixed):**
```python
train_dataset = (
    tf.data.Dataset.from_generator(train_generator, ...)
    .batch(per_replica_batch_size * num_replicas, drop_remainder=True)  # Correct
    ...
)
```

## Debugging Process

1. **dd35c4d**: Added diagnostic checks to identify NaN sources in RF training data
2. **3d7da8a**: Added more diagnostics to pinpoint the issue
3. **d2a38d6**: Identified and fixed the distributed dataset batch size bug
4. **c1ab0c8**: Reverted diagnostic code for final testing
5. **25a8cf5**: Renamed `global_batch_size` to `effective_batch_size` for clarity

## Impact

This bug affected any multi-GPU training run where `num_replicas > 1`. With 4 GPUs, each replica was only receiving 25% of the intended samples per step, leading to:
- Inconsistent gradient updates
- Potential numerical instability
- NaN values propagating through the model

## Files Changed
- `src/aetherscan/train.py`: Fixed batch size in dataset creation

## Test Plan
- [x] Verify fix by running multi-GPU training and confirming no NaN values
- [x] Confirm batch sizes match expectations in logs
- [ ] Validate training metrics are consistent across GPU counts
